mcan/src/config.rs

```diff
@@ -141,65 +160,102 @@ impl BitTiming {
+        /// frames.
+        data_phase_timing: BitTiming,
+    },
 }
 
-impl From<NonMatchingAction> for ANFS_A {
-    fn from(val: NonMatchingAction) -> Self {
-        match val {
-            NonMatchingAction::Fifo0 => ANFS_A::RXF0,
-            NonMatchingAction::Fifo1 => ANFS_A::RXF1,
-            NonMatchingAction::Reject => ANFS_A::REJECT,
-        }
+impl Default for Mode {
+    fn default() -> Self {
+        Self::Classic
     }
 }
 
-impl From<NonMatchingAction> for ANFE_A {
-    fn from(val: NonMatchingAction) -> Self {
-        match val {
-            NonMatchingAction::Fifo0 => ANFE_A::RXF0,
-            NonMatchingAction::Fifo1 => ANFE_A::RXF1,
-            NonMatchingAction::Reject => ANFE_A::REJECT,
+impl CanConfig {
+    pub(crate) fn new(bitrate: HertzU32) -> Self {
+        Self {
+            mode: Default::default(),
+            loopback: Default::default(),
+            nominal_timing: BitTiming::new(bitrate),
+            timestamp: Default::default(),
+            rx_fifo_0: Default::default(),
+            rx_fifo_1: Default::default(),
+            tx: Default::default(),
         }
     }
 }
 
-impl Default for NonMatchingAction {
+/// Denotes a RX-fifo configuration
+#[derive(Default, Copy, Clone)]
+pub struct RxFifoConfig {
+    /// FIFO mode
+    pub mode: RxFifoMode,
+    /// Fifo fullnes to generate interrupt
+    pub watermark: u8,
+}
+
+/// Operating modes for the two FIFO
+#[derive(Copy, Clone)]
+pub enum RxFifoMode {
+    /// Blocking mode
+    /// When the RX FIFO is full, not messages are written until at least one
+    /// has been read out
+    Blocking,
+    /// Overwriting mode
+    /// When the RX FIFO is full, the oldest messsage will be deleted and a new
+    /// message will take its place
+    Overwrite,
+}
+
+impl Default for RxFifoMode {
     fn default() -> Self {
-        Self::Reject
+        Self::Blocking
     }
 }
 
-/// Enable/disable CAN-FD and related features
-#[derive(Copy, Clone)]
-pub enum FdFeatures {
-    /// Classic mode with 8-bytes data. Reception of an FD frame is considered
-    /// an error.
-    ClassicOnly,
-    /// Transmission and reception of CAN FD frames (with up to 64 bytes of
-    /// data) is enabled. This does not prevent use of classic CAN frames.
-    Fd {
-        /// If `true`, FD frames can be transmitted with bit rate switching.
-        allow_bit_rate_switching: bool,
-        /// Bit timing parameters for the data phase of bit rate switched FD
-        /// frames.
-        data_phase_timing: BitTiming,
-    },
+impl From<RxFifoMode> for bool {
+    fn from(val: RxFifoMode) -> Self {
+        match val {
+            RxFifoMode::Overwrite => true,
+            RxFifoMode::Blocking => false,
+        }
+    }
 }
 
-/// Test modes for the bus
+/// How to treat the transmit buffer
 #[derive(Copy, Clone)]
-pub enum TestMode {
-    /// Do not initialize a test
-    Disabled,
-    /// Setup loopback
-    Loopback,
+pub enum TxBufferMode {
```

Comment:
How about `TxQueueMode`? The dedicated TX buffers are not affected by this mode selection.

Reply:
Responded in the previous conversation to this